artiq/frontend/artiq_flash.py

```diff
@@ -346,6 +348,9 @@ def main():
         action in args.action
         for action in ["gateware", "rtm_gateware", "bootloader", "firmware", "load", "rtm_load"])
     variant = args.variant
+    if variant is None and args.no_variant:
+        # Just use default actions in the below.
+        variant = ""
     if needs_artifacts and variant is None:
```

Comment:
This (variant listing) still gets executed when ``--no-variant`` is given. Is that OK?

Reply:
Is it? Variant should no longer be `None` then.

Give me ~1h to sort out the SiPyCo deployment so I can test this in situ…

Reply:
Works as intended; the variant listing is not executed.